benchmarks/torchvision/voirfile.py

```diff
@@ -1,5 +1,74 @@
 # Import this to instrument the ArgumentParser, remove if no such thing
 from milabench.opt import instrument_argparse
+from deepspeed.profiling.flops_profiler import FlopsProfiler
+import torch.profiler as profiler
+import torch
+
+
+class ModelCallWrapper:
+    """
+    Wrapper to replace model.__call__ method with a wrapper
+    which must profile each call.
+    """
+
+    def __init__(self, model):
+        cls = type(model)
+        self.model = model
+        self.original_call = cls.__call__
+        self.measures = []
+        cls.__call__ = self.call_torch_profiler
```

Comment:
Instead of replacing `__call__`, this can be done with ptera, in principle:

```python
# This part already exists and delimits a block
ov.probe(
    "//train_epoch(inp as batch, !#loop_inp as compute_start, !!#endloop_inp as compute_end)"
).give()

# This would profile the block delimited above
def _profile():
    with profiler.profile(...):
        yield
    return profiler.stats_and_whatever

ov.given.wmap("compute_start", _profile).give("profiler_stats")
```

The basic idea is that a selector with a !var and a !!var creates a sort of pseudo-context manager that lies between these two variables and is named after the !var. `#loop_x` and `#endloop_x` are meta-variables representing the beginning and end of a `for x in ...: ...` loop. `wmap` (the w stands for "wrap") can then be used to map a context manager around the block.

This has the advantage that:

1. The profiler will also profile the backward() pass, which I believe it will not if only `model.__call__` is wrapped.
2. We can reuse the `compute_start` probes that all benchmarks should aim to provide.

I'm not sure, but it might also solve the infinite recursion issue with deepspeed.